README.md

```diff
@@ -1 +1,588 @@
+
+Some questions that I often get during workshops, or from new colleagues who have just started working with LitElement are: 
+- How exactly does LitElement achieve its efficient updates? 
+- What does asynchronous rendering mean? 
+- How does LitElement pick up property changes?
+
+So I figured, _it's blog time!_
+
+Lets start by taking a look at the following code example:
+
+```js
+class MyElement extends LitElement {
+  foo() {
+    this.myPropertyA = 1;
+  }
+}
+```
+
+LitElement is _reactive_ and _observes_ your properties to kick off what is known as its "render pipeline". What this means is that whenever we change a property in our component, it'll update and re-render the component. This is nice, because it means we don't have to do anything manually. In the code example above, just the act of setting a new value to `this.myPropertyA` will cause our component to re-render.
+
+Now consider another example:
+```js
+class MyElement extends LitElement {
+  foo() {
+    this.myPropertyA = 1;
+    this.myPropertyB = 2;
+  }
+}
+```
+
+Hrm. Looks like we're in a bit of a pickle here. When should we re-render? Should we re-render after `myPropertyA` has been set? That would mean we might do a bunch of unnecessary work and rendering, because we also want our component to update when we set `myPropertyB`, so we'll end up with two renders, where one would have been sufficient. That doesn't sound very efficient. This would get out of hand really quickly if we'd have a lot more properties to set in a method.
+
+Fortunately, LitElement is smart and lazy, and uses a clever technique to _batch_ these updates, and only re-render _once_.
+
+## Batching
+
+Before we get started, lets make clear what we actually mean when we talk about _batching work_, and go over a couple techniques to demonstrate how we can achieve batching behavior in JavaScript.
+
+Batching is basically limiting the amount of work we need to do, and is often implemented as a performance measure. A popular usecase for batching, or limiting work, is limiting API calls that may get fired in quick succession for example. We don't want to put any unnecessary load on the API, and do a bunch of wasteful requests. Or in the case of LitElement; efficiently limiting re-renders of our components. Remember; the goal is to avoid _any_ unnecessary work in order to be performant.
+
+A popular pattern for batching is using a debounce function. As mentioned above, debouncers are often used for batching API calls. Imagine we have a search input, that every time a new search keyword is entered, fetches some search results from a server. It would put a pretty heavy load on the API if we would fire a request for every single input change or every new character, and that can be pretty wasteful. Instead, we can _debounce_ the API calls, to only actually fire a request on the _last_ input change.
+
+Consider the following example:
+
+```js
```

Comment:
Maybe instead of a code snippet, use a diagram to explain debounce, since it's not the main focus of the article, and code snippets (even stunningly beautiful ones like this) tend to tire the reader